src/Features/Core/Portable/EmbeddedLanguages/StackFrame/StackFrameNodeDefinitions.cs

```diff
@@ -0,0 +1,375 @@
+            : base(StackFrameKind.TypeIdentifier)
+        {
+            Identifier = identifier;
+        }
+
+        public override void Accept(IStackFrameNodeVisitor visitor)
+            => visitor.Visit(this);
+
+        internal override StackFrameNodeOrToken ChildAt(int index)
+            => index switch
+            {
+                0 => Identifier,
+                _ => throw new InvalidOperationException()
+            };
+
+        internal override StackFrameBaseIdentifierNode WithTrailingTrivia(StackFrameTrivia trailingTrivia)
+            => new StackFrameTypeArgument(Identifier.With(trailingTrivia: ImmutableArray.Create(trailingTrivia)));
+    }
+
+    internal sealed class StackFrameArgumentList : StackFrameNode
+    {
+        private readonly ImmutableArray<StackFrameNodeOrToken> _childNodesOrTokens;
+
+        public StackFrameArgumentList(ImmutableArray<StackFrameNodeOrToken> childNodesOrTokens) : base(StackFrameKind.ArgumentList)
+        {
+#if DEBUG
+            // The list should contain an open and close token, and optionally parameter definitions (each being type and name)
+            Debug.Assert(childNodesOrTokens.Length == 2 || childNodesOrTokens.Length >= 4);
+            for (var i = 0; i < childNodesOrTokens.Length; i++)
+            {
+                var nodeOrToken = childNodesOrTokens[i];
+
+                if (i == 0)
+                {
+                    Debug.Assert(nodeOrToken.Token.Kind is StackFrameKind.OpenParenToken);
+                    continue;
+                }
+
+                if (i == childNodesOrTokens.Length - 1)
+                {
+                    Debug.Assert(nodeOrToken.Token.Kind is StackFrameKind.CloseParenToken);
+                    continue;
+                }
+
+                if (nodeOrToken.IsNode)
+                {
+                    Debug.Assert(nodeOrToken.Node is StackFrameIdentifierNode or StackFrameMemberAccessExpressionNode or StackFrameArrayExpressionNode);
+                }
+                else
+                {
+                    Debug.Assert(nodeOrToken.Token.Kind is StackFrameKind.CommaToken or StackFrameKind.TextToken);
+                }
+            }
+#endif
+            _childNodesOrTokens = childNodesOrTokens;
+        }
+
+        internal override int ChildCount => _childNodesOrTokens.Length;
+
+        public override void Accept(IStackFrameNodeVisitor visitor)
+            => visitor.Visit(this);
+
+        internal override StackFrameNodeOrToken ChildAt(int index)
+            => _childNodesOrTokens[index];
+    }
+
+    internal sealed class StackFrameFileInformationNode : StackFrameNode
+    {
+        public StackFrameFileInformationNode(StackFrameKind kind) : base(kind)
+        {
+        }
+
+        internal override int ChildCount => throw new NotImplementedException();
+
+        public override void Accept(IStackFrameNodeVisitor visitor)
+        {
+            throw new NotImplementedException();
```

Comment:
this seems bad.

Reply:
https://github.com/dotnet/roslyn/pull/56957/files#r722733540